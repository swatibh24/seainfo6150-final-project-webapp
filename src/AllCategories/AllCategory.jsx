import React from "react";
import { Link } from "react-router-dom";
import styles from "./AllCategory.module.css";

const AllCategory = () => {
  return (
    <div className={styles.allCatContainer}>
      <h2 className={styles.allCatDetail}>Pick the categories you would like to explore!
      </h2>
    <div className={styles.ListContainer}>
     <div className={styles.container}>
      <Link className={styles.styleCategories} to="/category/Soup">
        <p className={styles.allCategoryTypes}>Soup</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Appetizer">
        <p className={styles.allCategoryTypes}>Appetizer</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Snack">
        <p className={styles.allCategoryTypes}>Snack</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Main-Course">
        <p className={styles.allCategoryTypes}>Main Course</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Dessert">
        <p className={styles.allCategoryTypes}>Dessert</p>
      </Link>
      </div>
      </div>
         <img className={styles.allCatImg} src="https://c0.wallpaperflare.com/preview/301/367/805/black-bean-close-up-cooked-cuisine.jpg" alt="categories"/>
    </div>
  );
};

export default AllCategory;
