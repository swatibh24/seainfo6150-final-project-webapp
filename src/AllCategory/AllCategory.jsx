import React from "react";
import { Link } from "react-router-dom";
import styles from "./AllCategory.module.css";

const AllCategory = () => {
  return (
    <div className={styles.allCategoryContainer}>
      <h2 className={styles.allCategoryDetail}>Pick the categories you would like to explore!
      </h2>
    <div className={styles.ListContainer}>
     <div className={styles.container}>
      <Link className={styles.styleCategories} to="/category/Soup">
        <p className={styles.allCategoryName}>Soup</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Appetizer">
        <p className={styles.allCategoryName}>Appetizer</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Snack">
        <p className={styles.allCategoryName}>Snack</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Main-Course">
        <p className={styles.allCategoryName}>Main Course</p>
      </Link>
      <Link className={styles.styleCategories} to="/category/Dessert">
        <p className={styles.allCategoryName}>Dessert</p>
      </Link>
      </div>
      </div>
         <img className={styles.allCategoryImg}
         src="https://c0.wallpaperflare.com/preview/301/367/805/black-bean-close-up-cooked-cuisine.jpg" alt="categories"/>
    </div>
  );
};

export default AllCategory;
