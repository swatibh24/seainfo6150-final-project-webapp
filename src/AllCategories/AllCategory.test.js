import React from "react";
import { render, fireEvent } from "@testing-library/react";
import AllCategory from "./AllCategory.jsx";
import { BrowserRouter, Link } from "react-router-dom";

it("renders the AllCategory component correctly", () => {
  const { container } = render(<BrowserRouter><AllCategory /></BrowserRouter>);
  expect(container).toMatchSnapshot();
});