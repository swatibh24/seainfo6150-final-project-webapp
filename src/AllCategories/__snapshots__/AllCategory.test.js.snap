// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the AllCategory component correctly 1`] = `
<div>
  <div
    class="allCatContainer"
  >
    <h2
      class="allCatDetail"
    >
      Pick the categories you would like to explore!
    </h2>
    <div
      class="ListContainer"
    >
      <div
        class="container"
      >
        <a
          class="styleCategories"
          href="/category/Soup"
        >
          <p
            class="allCategoryTypes"
          >
            Soup
          </p>
        </a>
        <a
          class="styleCategories"
          href="/category/Appetizer"
        >
          <p
            class="allCategoryTypes"
          >
            Appetizer
          </p>
        </a>
        <a
          class="styleCategories"
          href="/category/Snack"
        >
          <p
            class="allCategoryTypes"
          >
            Snack
          </p>
        </a>
        <a
          class="styleCategories"
          href="/category/Main-Course"
        >
          <p
            class="allCategoryTypes"
          >
            Main Course
          </p>
        </a>
        <a
          class="styleCategories"
          href="/category/Dessert"
        >
          <p
            class="allCategoryTypes"
          >
            Dessert
          </p>
        </a>
      </div>
    </div>
    <img
      alt="categories"
      class="allCatImg"
      src="https://c0.wallpaperflare.com/preview/301/367/805/black-bean-close-up-cooked-cuisine.jpg"
    />
  </div>
</div>
`;
